### PR TITLE
chore: MU03 deployment checks

### LINKED ISSUE #90: Write deployment checks for MU03 gov script

### PR DESCRIPTION
**Pull request by ninocomputer**
### Description

Added some deployment checks in order to verify that the MU03 governance script works as expected.

### Other changes

Removed some unused code in MU01 deployment checks

### Tested

Simulated gov proposals on alfajores and baklava. Checks work.

### Related issues

- Fixes #90 

### Backwards compatibility

n/a

### Documentation

n/a

### CODE REVIEW

File: script/upgrades/MU03/MU03Checks.sol [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+
+      // verify asset1 of the deployed pool against the config
+      if (pool.asset1 != poolConfigs[i].asset1) {
+        console2.log(
+          "The asset1 of deployed pool: %s does not match the expected asset1: %s.",
+          pool.asset1,
+          poolConfigs[i].asset1
+        );
+        revert("asset1 of pool does not match the expected asset1. See logs.");
+      }
```
Comment (bowd) [inline script/upgrades/MU03/MU03Checks.sol:188]: Let's also add the rest of the fields for shits and giggles:
- pricingModule
- spread
- referanceRateFeedID
- minimumReports
- referanceRateResetFrequency
- stablePoolResetSize

File: script/upgrades/MU03/MU03Checks.sol [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+        revert("Not all trading limits were configured correctly.");
+      }
+      if (poolConfigs[i].asset0_timeStep0 != limits.timestep0) {
+        console2.log("timestep0 for %s, %s was not set ❌", pool.asset0, pool.asset1);
+        revert("Not all trading limits were configured correctly.");
+      }
+      if (poolConfigs[i].asset0_timeStep1 != limits.timestep1) {
+        console2.log("timestep1 for %s, %s was not set ❌", pool.asset0, pool.asset1);
+        revert("Not all trading limits were configured correctly.");
+      }
```
Comment (bowd) [inline script/upgrades/MU03/MU03Checks.sol:239]: Let's also check the `flags` here, to make sure the limits are enabled, values can be set there but if the flag isn't correct it won't matter.

File: script/upgrades/MU03/MU03Checks.sol [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+          poolConfigs[i].referenceRateFeedID,
+          medianDeltaBreaker
+        );
+        if (!medianDeltaStatus) {
+          console2.log("MedianDeltaBreaker not enabled for rate feed %s", poolConfigs[i].referenceRateFeedID);
+          revert("MedianDeltaBreaker not enabled for all rate feeds");
+        }
+
+        if (poolConfigs[i].isValueDeltaBreakerEnabled) {
+          (, , bool valueDeltaStatus) = breakerBox.rateFeedBreakerStatus(
```
Comment (bowd) [inline script/upgrades/MU03/MU03Checks.sol:301]: nit: there's a `breakerBox.isBreakerEnabled` that can be used as well

File: script/upgrades/MU03/MU03Checks.sol [hunk 4]
```diff
@@ -0,0 +1,687 @@
...
+    swapcEURtoCELO();
+    swapCeloTocBRL();
+    swapcBrlToCELO();
+    swapBridgedUSDCTocUSD();
+    swapcUSDtoBridgedUSDC();
+    swapBridgedUSDCTocEUR();
+    swapcEURtoBridgedUSDC();
+    swapBridgedUSDCtocBRL();
+    swapcBRLtoBridgedUSDC();
+  }
```
Comment (bowd) [inline script/upgrades/MU03/MU03Checks.sol:425]: For all of the swaps, let's also verify that the amounts returned are within what we would consider "normal" ranges, not just that they execute.

**Review by bowd — CHANGES_REQUESTED** [review]
👍 Nice set of tests, left some comments to make it ✨, also a general comment, all your functions are `public`, this adds a bit of unnecessary complexity (and execution time) to the contract, let's use `internal` for all of the functions after `run`.

File: script/upgrades/MU03/MU03Checks.sol (hunk 3 above)
Comment (ninocomputer) [reply script/upgrades/MU03/MU03Checks.sol:301]: 👍🏼

**Review by nvtaveras — CHANGES_REQUESTED** [review]
some small refactor change requested but overall great great work! 🚀

File: script/upgrades/MU03/MU03.sol [hunk 5]
```diff
@@ -240,7 +241,7 @@ contract MU03 is IMentoUpgrade, GovernanceScript {
   /**
    * @notice This function creates the transactions to configure the Mento V1 Exchanges.
    */
-  function proposal_configureV1Exchanges() public {
+  function proposal_scaleDownReserveFraction() public {
```
[the remark this answers is not in the document]
Comment (ninocomputer) [reply script/upgrades/MU03/MU03.sol:244]: we chatted about this and tldr is that reserveFraction was different on mainnet and testnets, so we wanted to make it consistent by hardcoding the values from mainnet

**Review by bowd — APPROVED** [review]

**Review by nvtaveras — APPROVED** [review]
🚢

**Pull request merged**